.circleci/config.yml

```diff
@@ -0,0 +1,57 @@
+version: 2.1
+jobs:
+  node-dependencies:
+    docker:
+      - image: circleci/node:lts
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-npm-dependencies-{{ checksum "package-lock.json" }}
+      - run:
+          name: Install dependencies with the current package-lock
+          command: npm ci
+      - save_cache:
+          key: v1-npm-dependencies-{{ checksum "package-lock.json" }}
+          paths:
+            - "node_modules"
+  build:
+    docker:
+      - image: circleci/node:lts
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-npm-dependencies-{{ checksum "package-lock.json" }}
+      - run: npm run build
```

Comment:
Why do we need to build anything? If the loader is written to a level that Node and Jest support, you won't need Babel or a build step at all.

Reply:
@racoleman Two problems: 

If `index.js` is written as a ES Module, webpack uses [interpret](https://www.npmjs.com/package/interpret) under the hood, which uses `cjs-loader`. If you have babel set up locally, you can get away with `webpack.config.babel.js` but not every user will be digesting this with `babel`. 

Jest (because its mocking everyone 😆 ) uses its own [resolver](https://github.com/facebook/jest/blob/master/packages/jest-runtime/src/index.ts#L56) which doesn't support ES6 import statements.

I figured since we are using es6 all the way here, that it would make sense to use babel for the tests and use babel to transpile the dist directory to avoid these issue.

Reply:
That's because [Webpack loaders should not be written as ES modules ](https://webpack.js.org/contribute/writing-a-loader/)😉.

Now you might could work around using a build for Jest by using `babel-jest`, but I have to ask: is it worth it to add all this infrastructure just to use import statements in one file?

Reply:
> That's because [Webpack loaders should not be written as ES modules ](https://webpack.js.org/contribute/writing-a-loader/)😉.

I don't think that is implied anywhere. It actually implies the opposite quite a bit in the documentation, such as in [loader-utils][https://webpack.js.org/contribute/writing-a-loader/#loader-utilities], [loader-dependencies](https://webpack.js.org/contribute/writing-a-loader/#loader-utilities)

They also recommend using `babel-jest` for testing and having your module src be in ES6, but having a runtime that uses CommonJS. I figured the best way to accomplish that was to transpile the build 

> Now you might could work around using a build for Jest by using babel-jest, but I have to ask: is it worth it to add all this infrastructure just to use import statements in one file?

yes, ES6 is life! 😆  Honestly don't think it adds much but I understand the rationale